package/src/StreamCache.ts

```diff
@@ -357,6 +395,53 @@ export class StreamCache<
     return !!(clientData && channelsData);
   }
 
+  private async removeOlderImages(
+    oldChannelsData: ChannelStateAndDataInput<At, Ch, Co, Me, Re, Us>[],
+    newChannelsData: ChannelStateAndDataInput<At, Ch, Co, Me, Re, Us>[],
+  ) {
+    const oldChannelsMessagesMap = extractChannelMessagesMap(oldChannelsData);
+    const newChannelsMessagesMap = extractChannelMessagesMap(newChannelsData);
+
+    const removedChannels: string[] = [];
+    const removedMessages: { channelId: string; messageId: string }[] = [];
+
+    // Extract array of paths for removed channels and messages
+    Object.keys(oldChannelsMessagesMap).forEach((oldChannelId) => {
+      if (!newChannelsMessagesMap[oldChannelId]) {
+        removedChannels.push(oldChannelId);
+        return;
+      }
+
+      Object.keys(oldChannelsMessagesMap[oldChannelId]).forEach((oldMessageId) => {
+        if (!newChannelsMessagesMap[oldChannelId][oldMessageId]) {
+          removedMessages.push({ channelId: oldChannelId, messageId: oldMessageId });
+        }
+      });
+    });
+
+    await Promise.all(
+      removedChannels.map((channelId) =>
+        Promise.all([removeChannelAttachments(channelId), removeChannelAvatars(channelId)]),
+      ),
+    );
+
+    await Promise.all(
+      removedMessages.map(({ channelId, messageId }) =>
+        removeMessageAttachments(channelId, messageId),
+      ),
+    );
+  }
+
+  public async syncCacheAndImages() {
+    const oldChannelsData = await this.cacheInterface.getItem(STREAM_CHAT_CHANNELS_DATA);
+    await this.syncCache();
+    const newChannelsData = await this.cacheInterface.getItem(STREAM_CHAT_CHANNELS_DATA);
+
+    if (!oldChannelsData) return;
+
+    await this.removeOlderImages(oldChannelsData, newChannelsData || []);
```

Comment:
This call can be pretty expensive. I want to do some testing on it with a lot of images @vishalnarkhede

Reply:
I was thinking about not awaiting it in order to let the removal happen in background but that may lead to inconsistencies when loading images

Reply:
Yeah let's try without await.

Reply:
That may lead to a problem where the image component loads the cached image and in case it rerenders for some reason, the url wont change but the cached image wont exist anymore. did you manage to test it with a lot of images? any problem?